warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PrintingVisitor.java

```diff
@@ -337,552 +346,978 @@ private void childrenAccept(SimpleNode node, ParserVisitor visitor, Object data)
     }
 
     public Object visit(ASTJexlScript node, Object data) {
-        output.writeLine(data + node.toString());
-        childrenAccept(node, this, data + PREFIX);
+        if (maxTermsToPrint > 0) {
+            output.writeLine(data + node.toString());
+            maxTermsToPrint -= 1;
+            childrenAccept(node, this, data + PREFIX);
+        } else {
+            LOGGER.trace("maxTermsToPrint is exhausted. Ignoring: " + node);
+        }
```

Comment:
It looks like this code will still emit a log message for each node in the query that exceeds the threshold. Given that, I think I would prefer to see something like this (explanations for changes in-line)
```
int termsPrinted = 0;

...

public Object visit(node, data) {
    if (termsPrinted > maxTermsToPrint) {
        return null;  // return early without printing for each additional node visited after the threshold
    }

    termsPrinted++;

    if (termsPrinted > maxTermsToPrint) {
        LOGGER.trace("exceeded max terms to print threshold of " + maxTermsToPrint);  // just log the threshold, logging the whole node for a very large query could be less than constructive
    }

    output.writeLine(data + node.toString());
    childrenAccept(node, this, data + PREFIX);
    return null;
}
```

Reply:
Thanks, will clean this up.